plugins/janus_videoroom.c

```diff
@@ -4387,6 +4406,65 @@ static json_t *janus_videoroom_process_synchronous_request(janus_videoroom_sessi
 		json_object_set_new(response, "room", string_ids ? json_string(room_id_str) : json_integer(room_id));
 		json_object_set_new(response, "rtp_forwarders", list);
 		goto prepare_response;
+	} else if(!strcasecmp(request_text, "enable_recording")) {
+		JANUS_VALIDATE_JSON_OBJECT(root, record_parameters,
+			error_code, error_cause, TRUE,
+			JANUS_VIDEOROOM_ERROR_MISSING_ELEMENT, JANUS_VIDEOROOM_ERROR_INVALID_ELEMENT);
+		if(error_code != 0)
+			goto prepare_response;
+		json_t *record = json_object_get(root, "record");
+		int recording_active = json_is_true(record);
+		JANUS_LOG(LOG_VERB, "Enable Recording : %d \n", (recording_active ? 1 : 0));
+		/* Lookup room */
+		janus_mutex_lock(&rooms_mutex);
+		janus_videoroom *videoroom = NULL;
+		error_code = janus_videoroom_access_room(root, FALSE, TRUE, &videoroom, error_cause, sizeof(error_cause));
+		if(error_code != 0) {
+			JANUS_LOG(LOG_ERR, "Failed to access videoroom\n");
+			janus_mutex_unlock(&rooms_mutex);
+			goto prepare_response;
+		}
+		janus_mutex_lock(&videoroom->mutex);
+		janus_mutex_unlock(&rooms_mutex);
+		/* Set recording status */
+		videoroom->record = recording_active ? TRUE : FALSE;
+		/* Iterate over all participants */
+		gpointer value;
+		GHashTableIter iter;
+		g_hash_table_iter_init(&iter, videoroom->participants);
+		while (g_hash_table_iter_next(&iter, NULL, &value)) {
+			janus_videoroom_publisher *participant = value;
+			if(participant && participant->session) {
+				janus_mutex_lock(&participant->rec_mutex);
+				gboolean prev_recording_active = participant->recording_active;
+				participant->recording_active = recording_active;
+				JANUS_LOG(LOG_VERB, "Setting record property: %s (room %"SCNu64", user %"SCNu64")\n", participant->recording_active ? "true" : "false", participant->room_id, participant->user_id);
+				/* Do we need to do something with the recordings right now? */
+				if(participant->recording_active != prev_recording_active) {
+					/* Something changed */
+					if(!participant->recording_active) {
+						/* Not recording (anymore?) */
+						janus_videoroom_recorder_close(participant);
+					} else if(participant->recording_active && participant->sdp) {
+						/* We've started recording, send a PLI/FIR and go on */
+						janus_videoroom_recorder_create(
+							participant, strstr(participant->sdp, "m=audio") != NULL,
+							strstr(participant->sdp, "m=video") != NULL,
+							strstr(participant->sdp, "m=application") != NULL);
```

Comment:
I know this is in the original VideoRoom code, but looking at it now I think that's a broken way of detecting whether the publisher is sending that media. The stream may have been rejected, or be inactive. Anyway, just saying it out loud as a reminder to myself, nothing that should really concern you here.

Reply:
I will leave this then.